core/src/test/java/io/questdb/test/griffin/wal/WalPurgeJobTest.java

```diff
@@ -312,6 +316,119 @@ public void testOneSegment() throws Exception {
         });
     }
 
+    @Test
+    public void testPendingSegmentTasks() throws Exception {
+        assertMemoryLeak(() -> {
+            // Creating a table creates a new WAL with a first segment.
+            final String tableName = testName.getMethodName();
+            compile("create table " + tableName + " as (" +
+                    "select x, " +
+                    " timestamp_sequence('2022-02-24', 1000000L) ts " +
+                    " from long_sequence(5)" +
+                    ") timestamp(ts) partition by DAY WAL");
+            assertWalExistence(true, tableName, 1);
+            assertWalLockExistence(true, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            assertSegmentLockExistence(true, tableName, 0);
+            assertSegmentExistence(false, tableName, 1, 1);
+            assertSegmentLockExistence(false, tableName, 1);
+
+            // Create a second segment.
+            compile("alter table " + tableName + " add column sss string");
+            executeInsert("insert into " + tableName + " values (6, '2022-02-24T00:00:05.000000Z', 'x')");
+            assertWalExistence(true, tableName, 1);
+            assertSegmentExistence(true, tableName, 1, 0);
+            final File segment1DirPath = assertSegmentExistence(true, tableName, 1, 1);
+            assertSegmentExistence(false, tableName, 1, 2);
+            assertSegmentLockExistence(false, tableName, 2);
```

Comment:
assertSegmentLockExistence always check the wal1

Reply:
Well-spotted:

I'll add the missing arg in there too.
These functions should be generic and re-usable across all walIDs and segmentIDs.

Reply:
Done.
Reviewed and fixed all other similar functions.